### ISSUE TITLE
formRef.current.isSubmitting is not changing when called outside the form

### ISSUE THREAD

**Issue by maalej** [formium/formik#3180]
## Bug report
I submit the form with a button outside the form using ```formRef.current.submitForm()```.

### Current Behavior
The form is correctly submitted. Both the variables ```isSubmitting``` (provided by formik) and ```formRef.current.isSubmitting``` are updated to true, then to false once the submission is done. However when the variable ```formRef.current.isSubmitting``` is called outside the form, it never changes.

### Expected behavior
When the variable ```formRef.current.isSubmitting``` is called outside the form, it should be set to True, during the submission, then set to False once the form submission is done.

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 2.2.6
| React            | 16.14.0
| TypeScript       |
| Browser          | Chrome
| npm/Yarn         | npm 6.14.12
| Operating System | Debian 10

**Comment by johnrom** [comment]
The way refs work in React is that they update _once_ when a component is mounted. So for example, an HTMLInputElement will be assigned to inputRef.current. Now, an HTMLInputElement is a mutable object, so when the value of that input changes, the value of inputRef.current.value also changes.

Unfortunately, Formik is not actually a mutable or stable object at all. It is a collection of refs and states and stuff which together equals something somewhat equivalent for every render. submitForm() for example is the same function on every render, because we use `useEventCallback`, meaning a lot of formik is itself refs. So you can think of formik as an object which looks like:

```
{
  submitForm: ref,
  handleChange: ref,
  values: notARef,
  isSubmitting: notARef,
}
```

So basically, with `ref`, you'll never be able to see updates to FormikState (values, errors, isSubmitting, etc). You will, however, be able to access the stable API. In v3 we provide something called getState() which will allow you to access future state from the stable API, and I think we need to add useFormikEffect for exactly the use case you're describing, so you could subscribe to changes to state from a ref. It would look more like:

```ts
const formik = useRef(null);
const [isSubmitting, setIsSubmitting] = useFormikEffect(state => state.isSubmitting);

useEffect(() => {
  formik.addFormEffect(setIsSubmitting);
}, []);
```

In other words, it's not a great option atm and Formik's ref is currently more useful for its imperative methods than accessing its state.
